File: common/src/main/java/com/datastrato/gravitino/dto/responses/SecretResponse.java
/*
 * Copyright 2024 Datastrato Pvt Ltd.
 * This software is licensed under the Apache License version 2.
 */
package com.datastrato.gravitino.dto.responses;

import com.datastrato.gravitino.dto.secret.SecretDTO;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.common.base.Preconditions;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;
import org.apache.commons.lang3.StringUtils;

/** Represents a response for a secret. */
@Getter
@ToString
@EqualsAndHashCode(callSuper = true)
public class SecretResponse extends BaseResponse {

  @JsonProperty("secret")
  private final SecretDTO secret;

  /**
   * Constructor for SecretResponse.
   *
   * @param secret The secret data transfer object.
   */
  public SecretResponse(SecretDTO secret) {
    super(0);
    this.secret = secret;
  }

  /** Default constructor for SecretResponse. (Used for Jackson deserialization.) */
  public SecretResponse() {
    super();
    this.secret = null;
  }

  /**
   * Validates the response data.
   *
   * @throws IllegalArgumentException if the secret is not set.
   */
  @Override
  public void validate() throws IllegalArgumentException {
    super.validate();

    Preconditions.checkArgument(secret != null, "secret must not be null");
    Preconditions.checkArgument(
        StringUtils.isNotBlank(secret.name()), "secret 'name' must not be null and empty");
    Preconditions.checkArgument(
        StringUtils.isNotBlank(secret.value()), "secret 'value' must not be null and empty");
    Preconditions.checkArgument(
        StringUtils.isNotBlank(secret.type()), "secret 'target' must not be null and empty");
    Preconditions.checkArgument(secret.auditInfo() != null, "secret 'auditInfo' must not be null");
  }
}

File: clients/client-python/gravitino/dto/responses/secret_response.py
"""
Copyright 2024 Datastrato Pvt Ltd.
This software is licensed under the Apache License version 2.
"""

from dataclasses import dataclass, field

from dataclasses_json import config

from gravitino.dto.responses.base_response import BaseResponse
from gravitino.dto.secret_dto import SecretDTO


@dataclass
class SecretResponse(BaseResponse):
    """Represents a response for a secret."""

    _secret: SecretDTO = field(metadata=config(field_name="secret"))

    def validate(self):
        """Validates the response data.

        Raises:
            IllegalArgumentException if the secret name, type or audit is not set.
        """
        super().validate()

        assert self._secret is not None, "secret must not be null"
        assert (
            self._secret.name() is not None
        ), "secret 'name' must not be null and empty"
        assert self._secret.value() is not None, "secret 'value' must not be null"
        assert self._secret.type() is not None, "secret 'type' must not be null"
        assert self._secret.audit_info() is not None, "secret 'audit' must not be null"

    def secret(self) -> SecretDTO:
        return self._secret
